src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -46,17 +48,26 @@ private static String createUrl(PineconeClientConfig clientConfig) {
     }
 
     public void deleteIndex(String indexName) throws IOException {
-        try (Response response = client.newCall(buildRequest(DELETE, indexName, TEXT_PLAIN,null)).execute()) {
+        Request request = buildRequest(HTTP_METHOD_DELETE, indexName, TEXT_PLAIN,null);
+        try (Response response = client.newCall(request).execute()) {
             handleResponse(response);
         }
     }
 
     public void createIndex(CreateIndexRequest createIndexRequest) throws IOException {
         MediaType mediaType = MediaType.parse("application/json; charset=utf-8");
         RequestBody requestBody = RequestBody.create(createIndexRequest.toJson(), mediaType);
+        Request request = buildRequest(HTTP_METHOD_POST, EMPTY_RESOURCE_PATH, TEXT_PLAIN, requestBody);
+        try (Response response = client.newCall(request).execute()) {
+            handleResponse(response);
+        }
+    }
 
-        try (Response response = client.newCall(buildRequest(POST, EMPTY_RESOURCE_PATH, TEXT_PLAIN, requestBody)).execute()) {
+    public IndexMeta describeIndex(String indexName) throws IOException {
```

Comment:
Returning a user defined object "IndexMeta" as we've discussed in the past that it is preferable to do so.